rep-xxxx.rst

```diff
@@ -0,0 +1,268 @@
+
+
+Dual or Multi-arm robots
+------------------------
+
+This REP does not specify any special conventions for robots with multiple arms/groups or kinematic chains.
+The standard approach of prefixing joint and link names to ensure uniqueness of all frames in the (combined) frame hierarchy should be used to avoid collisions.
+See the `Example Frame Hierarchies`_ section for examples of this.
+
+
+Frame Authorities
+-----------------
+
+The frames described in this REP will typically be part of the static description of robot models encoded in urdfs or xacros.
+As such, the frame authority is expected to be an instance of ``robot_state_publisher``, but this is not required.
+In cases where (complicated) kinematics preclude the use of standard nodes, a specialised node capable of publishing the necessary frames could be used.
+
+
+Exceptions
+----------
+
+The scope of potential robotics software is too broad to require all ROS software to follow the guidelines of this REP.
+However, choosing different conventions should be well justified, well documented, and is discouraged.
+
+
+Example Frame Hierarchies
+=========================
+
+Single manipulator
+------------------
+
+The following shows an example frame hierarchy for a single serial manipulator with just the default tool frame::
+
+  base_link
+  ├ base
+  └ link_1
+    └ link_2
+      └..
+       └ link_N
+         ├ tool0
+         └ flange
+
+Single manipulator with EEF
+---------------------------
+
+The following shows an example frame hierarchy for a single serial manipulator with an EEF model attached to ``flange`` and an additional tool frame::
+
+  base_link
+  ├ base
+  └ link_1
+    └ link_2
+      └ ..
+        └ link_N
+          ├ tool0
+          ├ flange
+          │ └ eef_base_link
+          │   └ ..
+          │     └ eef_link_N
+          └ eef_tcp
+
+Note the ``eef_`` prefix on the links in the EEF subhierarchy to prevent name clashes.
+
+Dual manipulator
+----------------
+
+The following shows an example frame hierarchy for a work cell that consists of two manipulators::
+
+  base_link
+  ├ ..
+  ├ left_base_link
+  │ ├ left_base
+  │ └ left_link_1
+  │   └ left_link_2
+  │     └..
+  │      └ left_link_N
+  │        ├ left_tool0
+  │        └ left_flange
+  ├ ..
+  └ right_base_link
+    ├ right_base
+    └ right_link_1
+      └ right_link_2
+        └..
+         └ right_link_N
+           ├ right_tool0
+           └ right_flange
+
+
+Compliance
+==========
+
+This REP depends on and is compliant with REP 103 [#REP103]_, except where stated otherwise.
+
+
+Questions
+=========
+
+#. are there robots that do not have ``tool0`` defined as a child of ``flange`` (I seem to remember some brand having ``tool0`` defined relative to the last link/joint, which was not the flange)? If not: make ``tool0`` child of ``flange``. Otherwise keep ``tool0`` a sibling of ``flange``.
```

Comment:
same question as for the *Single manipulator with EEF* here.

Reply:
I recommend we derive `tool0` from `flange`, not `link_6`.  Arguments can be made both ways, but there are many cases when the same offset may be required from `link_6` in both cases.  Less duplication/error if that offset is only defined once.  This can simply be a recommendation, not a requirement, so that unusual cases could derive `tool0` directly from the last link.

Reply:
Ok, makes sense.